### PR TITLE
Add allowInsecureConnections flag

### LINKED ISSUE #12: how can we use http as well or https with self signed certificate
In the android we can use  ConnectionBuilder}that permits connecting to http links, and ignores certificates for https connections. 
is there any way to use http instead of https.
I am getting this error if I am trying to use http

E/AndroidRuntime( 4341): java.lang.RuntimeException: An error occurred while executing doInBackground()
E/AndroidRuntime( 4341): 	at android.os.AsyncTask$3.done(AsyncTask.java:353)
E/AndroidRuntime( 4341): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
E/AndroidRuntime( 4341): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
E/AndroidRuntime( 4341): 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
E/AndroidRuntime( 4341): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
E/AndroidRuntime( 4341): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
E/AndroidRuntime( 4341): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
E/AndroidRuntime( 4341): 	at java.lang.Thread.run(Thread.java:764)
E/AndroidRuntime( 4341): Caused by: java.lang.IllegalArgumentException: only https connections are permitted
 And can we use https with self signed certificate

### PR DESCRIPTION
**Pull request by shawnz**
Fixes #12 

I'm not too experienced with Flutter and I'm not sure this is the best way to implement this. Please give your feedback. Thanks, Shawn

### CODE REVIEW

File: lib/src/authorization_request.dart [hunk 1]
```diff
@@ -12,7 +12,8 @@ class AuthorizationRequest extends CommonRequestDetails
       Map<String, String> additionalParameters,
       String issuer,
       String discoveryUrl,
-      List<String> promptValues}) {
+      List<String> promptValues,
+      bool requireHttps}) {
```
Comment (MaikuB) [inline lib/src/authorization_request.dart]: Can you rename all instances of `requireHttps` on the Dart and Java side to something like `allowInsecureConnections`? You should also be able to provide a default value

**Comment by shawnz** [comment]
Hi there @MaikuB, I made the changes you requested. I've also set up a non-SSL proxy for demo.identityserver.io available here: http://172.104.217.88:1234/. You can see that when this IdP is used with allowInsecureConnections = true, it works fine. When allowInsecureConnections = false, an IllegalArgumentException is thrown with message "only https connections are permitted". Try it with the following configuration:

`String _clientId = 'native.code';
  String _redirectUrl = 'io.identityserver.demo:/oauthredirect';
  String _issuer = 'http://172.104.217.88:1234';
  String _discoveryUrl =
      'http://172.104.217.88:1234/.well-known/openid-configuration';
  List<String> _scopes = [
    'openid',
    'profile',
    'email',
    'offline_access',
    'api'
  ];

  AuthorizationServiceConfiguration _serviceConfiguration =
      AuthorizationServiceConfiguration(
          'http://172.104.217.88:1234/connect/authorize',
          'http://172.104.217.88:1234/connect/token');`

Also note that the Application element in the AndroidManifest.xml file for the demo app must have the following attribute added for this to work: android:usesCleartextTraffic="true"

**Comment by MaikuB** [comment]
Thanks I'll give it a spin

**Comment by MaikuB** [comment]
LGTM. There's a minor code styling thing I'll change in this PR and I'll be cleaning up some of the existing code I had as well. Will look to release this as a beta release first so I can see if others can also help with with testing since it had been requested before

**Pull request merged**